src/System.Net.Http/tests/FunctionalTests/HttpProtocolTests.cs

```diff
@@ -14,6 +15,176 @@ public class HttpProtocolTests : HttpClientTestBase
     {
         protected virtual Stream GetStream(Stream s) => s;
 
+        [Fact]
+        public async Task GetAsyncVersion10_SendsVersion10_Success()
+        {
+            await LoopbackServer.CreateServerAsync(async (server, url) =>
+            {
+                using (HttpClient client = CreateHttpClient())
+                {
+                    HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, url);
+                    request.Version = HttpVersion.Version10;
+
+                    Task<HttpResponseMessage> getResponseTask = client.SendAsync(request);
+                    Task<List<string>> serverTask =
+                        LoopbackServer.ReadRequestAndSendResponseAsync(server,
+                            $"HTTP/1.1 200 OK\r\nDate: {DateTimeOffset.UtcNow:R}\r\nContent-Length: 0\r\n\r\n",
+                            new LoopbackServer.Options { ResponseStreamWrapper = GetStream });
+
+                    await TestHelper.WhenAllCompletedOrAnyFailed(getResponseTask, serverTask);
+
+                    var requestLines = await serverTask;
+                    Assert.Equal($"GET {url.PathAndQuery} HTTP/1.0", requestLines[0]);
+                }
+            });
+        }
+
+        [Fact]
```

Comment:
These two tests are identical except for some data that could be parameterized.  How about making them a Theory instead?

Reply:
To help minimize the boilerplate, I'd suggest combining these two tests like this:
```C#
[SkipOnTargetFramework(TargetFrameworkMonikers.Uap)] // Uap does not support 1.0
[Fact]
public async Task GetAsync_RequestVersion10_Success() => GetAsync_RequestVersion_Success(new Version(1,0));

[Fact]
public async Task GetAsync_RequestVersion11_Success() => GetAsync_RequestVersion_Success(new Version(1,1));

private async Task GetAsync_RequestVersion_Success(Version version)
{
    ... // code of your test, setting request.Version to version, and using version.ToString(2) in the expected output
}
```